### ISSUE TITLE
Error raised in tern_worker when parsing HTML

### ISSUE THREAD

**Issue by K-Thompson** [adobe/brackets#11162]
Brackets v1.3

The first time I touch below HTML file, I get these errors:

```
NodeSocketTransport - start
Worker: Error thrown in tern_worker:undefined is not a function
TypeError: undefined is not a function
    at infer (file:///C:/Program%20Files%20(x86)/Brackets/www/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:1100:39)
    at file:///C:/Program%20Files%20(x86)/Brackets/www/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:987:9
    at Object.BinaryExpression (file:///C:/Program%20Files%20(x86)/Brackets/www/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:906:15)
```

The above can be raised with below (validated) HTML file. The problem are the comments immediately after the opening `<body>` tag. I'm having a problem with code hinting not working (or stopping to work) and I'm wondering if this error might be part of the problem.

```
<!doctype html>
<html>
	<head>
		<meta http-equiv="content-type" content="text/html; charset=utf-8" />
		
		<title>TVM</title>


	</head>
	<body>
<!--		<div class="container"  style="background-color:#ccc">-->
<!--
Bootstrap classes
-->
<!-- On rows -->
<!--
<tr class="active">...</tr>
<tr class="success">...</tr>
<tr class="warning">...</tr>
<tr class="danger">...</tr>
<tr class="info">...</tr>
-->
<!-- On cells (`td` or `th`) -->
<!--
<tr>
	<td class="active">...</td>
	<td class="success">...</td>
	<td class="warning">...</td>
	<td class="danger">...</td>
	<td class="info">...</td>
</tr>
-->
<p>This is valid HTML</p>
	</body>
</html>
```

See StackOverflow [How to Fix Adobe Brackets Code Completion and Hinting](http://stackoverflow.com/questions/30393112/how-to-fix-adobe-brackets-code-completion-and-hinting)

**Comment by nethip** [comment]
@prksingh

**Comment by nethip** [comment]
@prksingh Did you get a chance to look at this issue?

**Comment by K-Thompson** [comment]
@prksingh, @nethip, this issue looks to be fixed. Can't duplicate the problem in v1.4. However, code hinting an parsing valid JavaScript file is still a problem. I had v1.4 opened the first time for about 30 seconds and the dialog announcing that a file can't be parsed opened. Perhaps this won't continue to happen with v1.4, but yesterday, with v1.3, while working on the same project, throughout the afternoon, I kept getting interrupted with the problem parsing dialog. It's not the interruption that is the problem. Rather, I don't understand why Brackets seems to still be parsing unopened and unused files within the project folder. I would think that the parsing should happen when Brackets opens and then be down with it.

**Comment by MarcelGerber** [comment]
I still get the `Error thrown in tern_worker:undefined is not a function` message in Release 1.4

Luckily, I just tried it with my PR #11569 and it looks like it fixes this case, too.

**Comment by MarcelGerber** [comment]
This is fixed with the merge of #11569 and will come to you in Release 1.8.
Please let us know if it still doesn't work for you when the Release ships.
